### PR TITLE
cql3: restrictions: remove primary_key_restrictions::bound_ranges()

### PR DESCRIPTION
**Pull request by michoecho**
Remove and/or document dead code.

### CODE REVIEW

File: cql3/restrictions/single_column_primary_key_restrictions.hh [hunk 1]
```diff
@@ -307,7 +307,12 @@ private:
     }
 
 public:
-    std::vector<bounds_range_type> bounds_ranges(const query_options& options) const override;
+    // When ValueType is clustering_key_prefix, this overrides a method of clustering_key_restrictions.
+    // When ValueType is partition_key, this method is unused. That's why we can't add `override`
+    // to the declaration.
+    // It would be great if we could conditionally implement this method only when
+    // ValueType is partition_key, but this seems to be impossible in C++.
```
Comment (haaawk) [inline cql3/restrictions/single_column_primary_key_restrictions.hh:314]: Did you mean `when ValueType is clustering_key_prefix`?

**Review by haaawk — APPROVED** [review]
LGTM

File: cql3/restrictions/single_column_primary_key_restrictions.hh (hunk 1 above)
Comment (michoecho) [reply cql3/restrictions/single_column_primary_key_restrictions.hh]: Yes, that's what I meant. Thanks. Fixed.

File: cql3/restrictions/primary_key_restrictions.hh [hunk 2]
```diff
@@ -76,8 +76,6 @@ public:
         return this->shared_from_this();
     }
 
-    virtual std::vector<bounds_range_type> bounds_ranges(const query_options& options) const = 0;
-
     using restrictions::has_supporting_index;
```
Comment (avikivity) [inline cql3/restrictions/primary_key_restrictions.hh:79]: Why is this removed? Should go into changelog.

File: cql3/restrictions/single_column_primary_key_restrictions.hh [hunk 3]
```diff
@@ -355,22 +360,12 @@ public:
...
-        if (!r.is_singular()) {
-            throw exceptions::invalid_request_exception("Range queries on partition key values not supported.");
-        }
-        ranges.emplace_back(std::move(r).transform(
-            [this] (partition_key&& k) -> query::ring_position {
-                auto token = dht::get_token(*_schema, k);
-                return { std::move(token), std::move(k) };
-            }));
-    }
-    return ranges;
```
Comment (avikivity) [inline cql3/restrictions/single_column_primary_key_restrictions.hh:373]: Why is this dead code? because of the expr conversions? -> changelog.

Comment (michoecho) [reply cql3/restrictions/single_column_primary_key_restrictions.hh:373]: I don't know for sure when this code died, but it seems that it happened about two years ago in 4bd0cb8dd9309237c8e107e86432c09cd2fec9ca and related patches.

There are `// FIXME(sarna): transitive hack only, do not judge. Should be dropped after all primary_key_restrictions<T> uses are removed from code.` and `// NOTICE(sarna): This function is useless for partition key restrictions, but it should remain here until single_column_primary_key_restrictions class is detemplatized.` scattered around. It seems that some refactoring was unfinished and some dead code remained.

I only noticed it because `token_restriction::bounds_ranges` contains some token-related code, which I carefully rewrote only to realize that it's not called anywhere.

Comment (michoecho) [reply cql3/restrictions/single_column_primary_key_restrictions.hh:373]: @psarna Do you remember this? What are those leftovers?

Comment (psarna) [reply cql3/restrictions/single_column_primary_key_restrictions.hh:373]: I don't remember exactly, but it's entirely possible that this code died when I partially detemplatized the restrictions. The effort was later dropped because it was decided that this whole layer is going to be rewritten with expressions anyway.

**Comment by michoecho** [comment]
Closing for now. If this dead code is a result of an unfinished greater task, we should revisit this when we are ready to finish it.
